java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/routes/RouteRelationExtractor.java

```diff
@@ -0,0 +1,603 @@
+import net.osmand.data.LatLon;
+import net.osmand.gpx.GPXFile;
+import net.osmand.gpx.GPXUtilities;
+import net.osmand.impl.ConsoleProgressImplementation;
+import net.osmand.obf.OsmGpxWriteContext;
+import net.osmand.obf.preparation.DBDialect;
+import net.osmand.obf.preparation.OsmDbAccessor;
+import net.osmand.obf.preparation.OsmDbAccessorContext;
+import net.osmand.obf.preparation.OsmDbCreator;
+import net.osmand.osm.MapRenderingTypesEncoder;
+import net.osmand.osm.RelationTagsPropagation;
+import net.osmand.osm.edit.Entity;
+import net.osmand.osm.edit.Entity.EntityId;
+import net.osmand.osm.edit.Node;
+import net.osmand.osm.edit.Relation;
+import net.osmand.osm.edit.Way;
+import net.osmand.osm.io.OsmBaseStorage;
+import net.osmand.osm.io.OsmBaseStoragePbf;
+import net.osmand.osm.io.OsmStorageWriter;
+import net.osmand.render.FindByRenderingTypesRules;
+import net.osmand.shared.io.KFile;
+import net.osmand.util.Algorithms;
+import net.osmand.util.MapUtils;
+import org.apache.commons.compress.compressors.bzip2.BZip2CompressorInputStream;
+import org.apache.commons.compress.compressors.bzip2.BZip2CompressorOutputStream;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.xmlpull.v1.XmlPullParserException;
+
+import javax.xml.stream.XMLStreamException;
+import java.io.*;
+import java.nio.file.Files;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.*;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+import static net.osmand.IndexConstants.GPX_GZ_FILE_EXT;
+import static net.osmand.gpx.GPXUtilities.OSMAND_EXTENSIONS_PREFIX;
+import static net.osmand.gpx.GPXUtilities.writeNotNullText;
+import static net.osmand.obf.OsmGpxWriteContext.OSM_TAG_PREFIX;
+import static net.osmand.router.RouteExporter.OSMAND_ROUTER_V2;
+
+public class RouteRelationExtractor {
+//	private boolean DEBUG = false;
+	private static final Log log = LogFactory.getLog(RouteRelationExtractor.class);
+	int countFiles;
+	int countWays;
+	int countNodes;
+	DBDialect osmDBdialect = DBDialect.SQLITE;
+	private final double precisionLatLonEquals = 1e-5;
+
+	private String[] customStyles = {
+			"default.render.xml",
+			"routes.addon.render.xml"
+			// "skimap.render.xml" // could work instead of default.render.xml but not together
+	};
+	private static final Map<String, String> customProperties = Map.of(
+			// routes.addon.render.xml
+			"showCycleRoutes", "true",
+			"showMtbRoutes", "true",
+			"hikingRoutesOSMC", "walkingRoutesOSMC",
+			"showDirtbikeTrails", "true",
+			"horseRoutes", "true",
+			"showFitnessTrails", "true",
+			"showRunningRoutes", "true",
+			"whiteWaterSports", "true" // default.render.xml
+			// "pisteRoutes", "true" // skimap.render.xml conflicts with default
+	);
+	private FindByRenderingTypesRules finder = new FindByRenderingTypesRules(customStyles, customProperties);
+	private final String[] filteredTags = {
```

Comment:
exclude "-" tags

Reply:
The filter list is actualized, multiple route type tags like "hiking;bicycle" is supported, OsmAnd-Java convertFromOsmGPXTag is refilled with fresh OSM route tags (tested on Germany, Netherlands, Czech).